influxdb/1.5/init-influxdb.sh

```diff
@@ -11,8 +11,9 @@ fi
 
 INIT_USERS=$([ ! -z "$AUTH_ENABLED" ] && [ ! -z "$INFLUXDB_ADMIN_USER" ] && echo 1 || echo)
 
-if ( [ ! -z "$INIT_USERS" ] || [ ! -z "$INFLUXDB_DB" ] || [ "$(ls -A /docker-entrypoint-initdb.d 2> /dev/null)" ] ) && [ ! "$(ls -d /var/lib/influxdb/meta 2>/dev/null)" ]; then
+if [ -z $INFLUXDB_META_DIR ]; then META_DIR="/var/lib/influxdb/meta" else META_DIR=$INFLUXDB_META_DIR fi
 
+if ( [ ! -z "$INIT_USERS" ] || [ ! -z "$INFLUXDB_DB" ] || [ "$(ls -A /docker-entrypoint-initdb.d 2> /dev/null)" ] ) && [ ! "$(ls -d $META_DIR 2>/dev/null)" ]; then
```

Comment:
I think the `$META_DIR` needs to be encapsulated in quotes to avoid any problems with spaces.

```
jsternberg-laptop:flux jsternberg$ meta_dir="/usr/lib/influxdb/meta  dir"
jsternberg-laptop:flux jsternberg$ echo $meta_dir
/usr/lib/influxdb/meta dir
jsternberg-laptop:flux jsternberg$ echo "$meta_dir"
/usr/lib/influxdb/meta  dir
jsternberg-laptop:flux jsternberg$ listing="$(ls $meta_dir)"
ls: /usr/lib/influxdb/meta: No such file or directory
ls: dir: No such file or directory
jsternberg-laptop:flux jsternberg$ listing="$(ls "$meta_dir")"
ls: /usr/lib/influxdb/meta  dir: No such file or directory
```

As an explanation of the above, if you have spaces in your variable and you send it to echo, it will split on spaces and send each as a separate argument. `echo` always prints one space so if there are multiple spaces you won't see them. But, when you use quotes, the entire thing gets passed as one argument. Then I tried it using the subshell. It appears like you do not need to escape the quotes inside of the subshell because the subshell runs in a completely separate context.

So I think you just need to change it to `"$(ls -d "$META_DIR" 2> /dev/null)"`.

Reply:
sure